### PR TITLE
Integrate from Piper for C++, Java, and Python

### PR DESCRIPTION
**Pull request by fowles**
**Important:** this integration must be merged, not squashed!

### CODE REVIEW

File: cmake/extract_includes.bat.in [hunk 1]
```diff
@@ -0,0 +1,144 @@
+mkdir include
```
Comment (deannagarcia) [inline cmake/extract_includes.bat.in:1]: @mkruskal-google deleted this file in #10306, you should sync with him on whether to sync this back in/how to delete it from our sync process.

**Review by deannagarcia — COMMENTED** [review]
Any changes with proto relnotes to add to CHANGES.txt?

File: cmake/extract_includes.bat.in (hunk 1 above)
Comment (mkruskal-google) [reply cmake/extract_includes.bat.in:1]: Yea I wonder why it didn't sync the deletion back to google3 in Sandy's sync?  We should delete this here and in google3

**Review by mkruskal-google — CHANGES_REQUESTED** [review]
Please revert extract_includes.bat.in, I'll remove it from google3 so we don't keep hitting this

File: cmake/extract_includes.bat.in (hunk 1 above)
Comment (fowles) [reply cmake/extract_includes.bat.in]: Done.

**Comment by fowles** [comment]
@mkruskal-google PTAL

**Review by mkruskal-google — APPROVED** [review]

**Pull request merged**